drafts/use-cases/2.api-documentation.md

```diff
@@ -0,0 +1,111 @@
+                        "@id": "eventName"
+                    }
+                },
+                {
+                    "property": {
+                        "@id": "eventDescription"
+                    }
+                },
+                {
+                    "property": {
+                        "@id": "startDate"
+                    }
+                },
+                {
+                    "property": {
+                        "@id": "endDate"
+                    }
+                }
+            ]
+        }
+    ]
+}
+```
+
+In order to i.e. generate views for each supported class, a complete type definition should be provided.
+Currently, Hydra Core Vocabulary allows that using RDFS/OWL vocabularies.
+
+The documentation should also contain some human readable details like:
+- operation expected type and it's role
+- operation returned type and it's role
+- operation name
+- operation description
+- parameters and their descriptions
+
+### Considerations
+
+#### Entry-Point/API Documentation fetching precedence
+In the previous scenario, application fetched an entry-point data as it's first action.
+We also could consider an opposite - taking an API documentation if available.
+This would require some standardized mechanism of accesing that documentation.
+
+Few possibilities here are available:
+- sending an _OPTIONS_ HTTP request to the base URL,
+  but HTTP specification does not allow to cache these responses
+- sending an _GET_ HTTP request to the base URL with some HTTP header;
+  header _Expect_ would be a candidate, but it's not allowed in AJAX communication;
+  another option would be a _Prefer_
+
+Once the API documentation is fetched and processed,
+client would then display some base view with further options,
+i.e. a main menu with all supported classes.
+
+#### Data structure description
+As already mentioned, Hydra Core Vocabulary allows to embed class description using RDFS/OWL constructs.
+Unfortunately, using those may prove difficult in parsing.
+In the scenario above, we'd need to use _rdfs:range_ predicate to denote the types a given property accepts.
+For simple scenarios it may be just _xsd:string_, but what if server will come out with a union of classes?
+
+#### Embedding class supported operations
+We could embed within each of the supported class details on what can be done with those classes,
+i.e. how can we obtain all instances or how we can create a new one.
+This may be an alternative to creating i.e. main menu in the UI application -
+user will be presented all possibilities, from which some may be denied.
+In case of an entry-point and embedded hypermedia controls I'd expect only those that are allowed in the current state.
+Currently, Hydra Core Vocabulary does not give a clear answer on how such a construct.
+
+#### API documentation and hypermedia controls precedence
+It is unclear on how details obtain from API documentation and current hypermedia controls should be "merged".
```

Comment:
This is actually not unclear but very well defined due to being based on RDF whose biggest strength is data integration. You just add union the information.